client/src/components/DataFiles/DataFilesProjectMembers/_cells/SystemRoleSelector.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useEffect, createContext, useContext } from 'react';
+import { useQuery, useMutation } from 'react-query';
+import { useSelector } from 'react-redux';
+import Cookies from 'js-cookie';
+import fetch from 'cross-fetch';
+import DropdownSelector from '_common/DropdownSelector';
+import { Button } from 'reactstrap';
+import styles from '../DataFilesProjectMembers.module.scss';
+import LoadingSpinner from '_common/LoadingSpinner';
+
+const getSystemRole = async (projectId, username) => {
+  const url = `/api/projects/${projectId}/system-role/${username}/`;
+  const request = await fetch(url, {
+    headers: { 'X-CSRFToken': Cookies.get('csrftoken') },
+    credentials: 'same-origin',
+  });
+  const data = await request.json();
+  return data;
+};
+
+const setSystemRole = async (projectId, username, role) => {
+  const url = `/api/projects/${projectId}/members/`;
+  const request = await fetch(url, {
+    headers: { 'X-CSRFToken': Cookies.get('csrftoken') },
+    credentials: 'same-origin',
+    method: 'PATCH',
+    body: JSON.stringify({
+      action: 'change_system_role',
+      username,
+      newRole: role,
+    }),
+  });
+  const data = await request.json();
+  return data;
+};
+
+const useSystemRole = (projectId, username) => {
+  const query = useQuery(['system-role', projectId, username], () =>
+    getSystemRole(projectId, username)
+  );
+  const mutation = useMutation(async (role) => {
+    await setSystemRole(projectId, username, role);
+    query.refetch();
+  });
+  return { query, mutation };
+};
+
+const SystemRoleSelector = ({ projectId, username }) => {
+  const authenticatedUser = useSelector(
+    (state) => state.authenticatedUser.user.username
+  );
+  const { query: authenticatedUserQuery } = useSystemRole(
+    projectId,
+    authenticatedUser
+  );
+  const currentUserRole = authenticatedUserQuery.data?.role;
+
+  const {
+    query: { data, isLoading, isFetching, error },
+    mutation: { mutate: setSystemRole, isLoading: isMutating },
+  } = useSystemRole(projectId, username);
+
+  const [selectedRole, setSelectedRole] = useState(data?.role);
+  useEffect(() => setSelectedRole(data?.role), [data?.role]);
```

Comment:
Changing the system role does not change the project role. Did we decide to forego messing with "project" roles?

Reply:
Yeah, my understanding was that we're getting rid of the co-PI/member distinction since it's a completely portal-side abstraction and we don't *do* anything with it.

Reply:
Gotcha. As a part of this PR, project role functions are expanded - do we want to make note of that somewhere?

Reply:
As in a note in the UI, or in our internal docs?

Reply:
Was just thinking a line comment. Just something that mentions that we don't expect to use those features